traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -388,7 +388,7 @@ func createV50(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, ds tc.D
 		return nil, http.StatusInternalServerError, nil, sysErr
 	}
 	if authorized, err := isTenantAuthorized(inf, &ds); err != nil {
-		return nil, http.StatusInternalServerError, nil, fmt.Errorf("checking tenant: %w", err)
+		return nil, http.StatusBadRequest, err, fmt.Errorf("checking tenant: %w", err)
```

Comment:
This exposes error information from connections with the database to API clients. One of two things should instead be done:
* check here that the error returned by `isTenantAuthorized` does/doesn't represent a failure to find any matching rows and handle appropriately (404 if yes, 500 otherwise)
* refactor `isTenantAuthorized` to instead return a third value which indicates whether any such Tenant could be found a la [`dbhelpers.GetStatusByID`](https://pkg.go.dev/github.com/apache/trafficcontrol@v7.0.1+incompatible/traffic_ops/traffic_ops_golang/dbhelpers#GetStatusByID) (and others). More work, but a better solution IMO.